### ISSUE TITLE
Identify any commonly added rules which are not part of this set

### ISSUE THREAD

**Issue by edwilde** [silverstripeltd/bespoke-standards#6]
This ruleset is traditionally a _starting point_ for new projects. Because of this many project implement their own customisations, some projects often implement the same customisation.

Ideally we would have a script which would parse all `silverstripeltd` repos, extract the rules from the root `phpcs.xml` and identify any rules which are common to many repositories and missing from this set.

This could potentially be run periodically to help keep this module up to date with the customisations added in the future.

**Comment by n8-dev** [comment]
Oooh.

This sounds cool, def be keen to work on this in PDP time.

Is this something github actions could do?
Create an ISSUE when it finds a divergence?


Many thoughts here.

**Comment by edwilde** [comment]
I think it would probably be a bit of local github api work, it could probably be done with an action but the action would need access to all the private repos.

**Comment by n8-dev** [comment]
Heh, I was just thinking about how one could build a dashboard for multi repos and did a cursory peak at the github API.

Let the ideas percolate and form! 💡

**Comment by TheSceneman** [comment]
Was thinking of building a module which leverages this library
https://github.com/KnpLabs/php-github-api/

And runs github api stuff as dev tasks from within SS framework

**Comment by TheSceneman** [comment]
@edwilde @n8-dev Have assembled a build task to perform this check. I have invited you both to a private repo housing the script.

Heres what it output:
**Common sniffs found:**
PSR12
Generic.Arrays.DisallowLongArraySyntax

**New sniffs found:**
Generic.Arrays.DisallowLongArraySyntax
./vendor/silverstripeltd/bespoke-standards/ruleset.xml
PSR12
Generic.Files.LineLength
SlevomatCodingStandard.Namespaces.ReferenceUsedNamesOnly
SlevomatCodingStandard.Files.LineLength
Squiz.Strings.ConcatenationSpacing
SlevomatCodingStandard.TypeHints.PropertyTypeHint.MissingAnyTypeHint
SlevomatCodingStandard.TypeHints.ReturnTypeHint.MissingNativeTypeHint
SlevomatCodingStandard.Namespaces.ReferenceUsedNamesOnly.ReferenceViaFullyQualifiedName
SlevomatCodingStandard.Classes.ClassStructure

"new sniffs" indicates rules which were found in other phpcs.xml files but are not included in the bespoke standards ruleset. Common sniffs are sniffs which are are in more than one repo.

Link to file is here https://github.com/TheSceneman/silverstripe-github-tasks/blob/5/src/CompareSniffsTask.php

**Comment by edwilde** [comment]
Wow, nice work...

`Generic.Arrays.DisallowLongArraySyntax` is certainly one which should be added imo, although I am surprised there isn't more deviation.

**Comment by TheSceneman** [comment]
I was surprised as well, made me second guess the code a bit... I feel like the results are correct though. Theres a lot of output on the job and I'm confident every repo was examined.

**Comment by satrun77** [comment]
I was thinking of implementing that sniff `DisallowLongArraySyntax`. We should add it

**Comment by TheSceneman** [comment]
Good call, it is a good sniff! No one likes `array()`! Was interested in the history of this, found the square bracket notation was added as part of PHP 5.4 in 2012. We should enforce this.

I will open a pull request

**Comment by n8-dev** [comment]
Yeahhhh @TheSceneman 👏 


Will give your code a hoon when I can.
Def surprised that is it 😆

Then again...
Over time with the update being a copy... it kindaaaa makes sense...
